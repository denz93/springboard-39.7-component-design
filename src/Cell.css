.Cell {
  vertical-align: middle;
  height: 100px;
  width: 100px;
  /* background-color: darkgray; */
  cursor: pointer;
  border: 1px dashed black;
}
.Cell:hover {
  background-color: aliceblue;
  ;
}

.Cell:hover img {
  transition: scale 0.2s ease-in-out;
  scale: 1.2;
}

