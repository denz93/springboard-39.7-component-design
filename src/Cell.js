import React from "react";
import "./Cell.css";
import lightOn from './light-on.png';
import lightOff from './light-off.png';

/** A single cell on the board.
 *
 * This has no state --- just two props:
 *
 * - flipCellsAroundMe: a function rec'd from the board which flips this
 *      cell and the cells around of it
 *
 * - isLit: boolean, is this cell lit?
 *
 * This handles clicks --- by calling flipCellsAroundMe
 *
 **/

function Cell({ flipCellsAroundMe, isLit }) {
  const classes = `Cell ${isLit ? "Cell-lit" : ""}`;

  return <td className={classes} onClick={flipCellsAroundMe} >
    <img src={isLit ? lightOn : lightOff} alt={isLit ? "light on" : "light off"} width={64} height={64}/>
  </td>;
}

export default Cell;
