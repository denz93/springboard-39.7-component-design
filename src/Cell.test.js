import {render} from '@testing-library/react'
import Cell from './Cell';

it('renders without crashing', () => {
  render(<Cell />)
})

it('should match unlit snapshot', () => {
  const {asFragment} = render(<Cell />)
  expect(asFragment()).toMatchSnapshot()
})
it('should match lit snapshot', () => {
  const {asFragment} = render(<Cell isLit={true} />)
  expect(asFragment()).toMatchSnapshot()
})