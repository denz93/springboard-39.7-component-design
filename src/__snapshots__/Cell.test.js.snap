// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match lit snapshot 1`] = `
<DocumentFragment>
  <img
    alt="light on"
    height="64"
    src="light-on.png"
    width="64"
  />
</DocumentFragment>
`;

exports[`should match unlit snapshot 1`] = `
<DocumentFragment>
  <img
    alt="light off"
    height="64"
    src="light-off.png"
    width="64"
  />
</DocumentFragment>
`;
